rpc-broker/src/policy.c

```diff
@@ -0,0 +1,258 @@
+/*
+ Copyright (c) 2018 AIS, Inc.
+
+ This program is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program; if not, write to the Free Software
+ Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+*/
+
+#include "rpc-broker.h"
+
+
+static int create_rule(struct rule *current, char *rule)
+{
+    char *token;
+    const char *delimiter;
+
+    if (!rule || rule[0] == '\0')
+        return -1;
+
+    current->rule_string = strdup(rule);
+
+    delimiter = " ";
+    token = strtok(rule, delimiter);
+
+    if (!token)
+        return -1;
+
+    current->policy = token[0] == 'a';
+    token = strtok(NULL, delimiter);
+
+    while (token) {
+
+        char *field = strtok(NULL, delimiter);
+
+        if (!field && token[0] != 's') {
+            return -1;
+        } else if (!strcmp("destination", token)) {
+            current->destination = strdup(field);
+        } else if (!strcmp("dom-type", token)) {
+            current->domtype = strdup(field);
+        } else if (!strcmp("interface", token)) {
+            current->interface = strdup(field);
+        } else if (!strcmp("if-boolean", token)) {
+            current->if_bool = strdup(field);
+            token = strtok(NULL, delimiter);
+            current->if_bool_flag = strcmp("true", token) == 0 ? 1 : 0;
```

Comment:
Again, super hard to read, also `strcmp("true", token) == 0` is better written `!strcmp("true", token)`.